staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -314,6 +363,47 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 					defer proxy.Close()
 				}
 
+				var l net.Listener
+				if testCase.socks5ProxyServerFunc != nil {
+					var proxyHandler *socks5.Server
+					if testCase.proxyAuth != nil {
+						proxyHandler = socks5Server(&socks5.StaticCredentials{
+							"proxyuser": "proxypasswd",
+						})
+						proxyCalledWithAuth = true
+					} else {
+						proxyHandler = socks5Server(nil)
+						proxyCalledWithAuth = false
+					}
+
+					l, err = net.Listen("tcp", "127.0.0.1:0")
+					if err != nil {
+						t.Fatalf("socks5Server: proxy_test: Listen: %v", err)
+					}
+					defer l.Close()
+
+					proxyCalledWithHost = "//127.0.0.1:" + strconv.Itoa(l.Addr().(*net.TCPAddr).Port)
+					go proxyHandler.Serve(l)
+					time.Sleep(100 * time.Millisecond)
```

Comment:
is there a more robust way to make sure the server is ready before we continue?

Reply:
@liggitt: do you have recommendations for an alternative strategy?